# HG changeset patch
# User Shai Ayal <shaiay@users.sourceforge.net>
# Date 1205089418 -7200
#      Sun Mar 09 21:03:38 2008 +0200
# Node ID dfae35ac4fb03ff2a57ac48aab928acb8fb9c666
# Parent  09b1a9c8812850cd4ba12b1b52b7698697ed16b4
added zoomstack support to fltk_backend

diff --git a/src/graphics/ChangeLog b/src/graphics/ChangeLog
--- a/src/graphics/ChangeLog
+++ b/src/graphics/ChangeLog
@@ -1,8 +1,13 @@
+2008-03-09  Shai Ayal  <shaiay@users.sourceforge.net>
+
+	* fltk_backend/fltk_backend.cc (plot_window::handle): add zoom
+	stack
+
 2008-03-06  Shai Ayal  <shaiay@users.sourceforge.net>
 
 	* fltk_backend/fltk_backend.cc (plot_window::handle): Add handling
 	of the 'a' and 'g' keys
 	(plot_window: toggle_grid): New helper function
 	(plot_window): Add new togglegrid button
 
 2008-03-01  Shai Ayal  <shaiay@users.sourceforge.net>
diff --git a/src/graphics/fltk_backend/fltk_backend.cc b/src/graphics/fltk_backend/fltk_backend.cc
--- a/src/graphics/fltk_backend/fltk_backend.cc
+++ b/src/graphics/fltk_backend/fltk_backend.cc
@@ -427,54 +427,62 @@ private:
 		graphics_object ax = 
 		  gh_manager::get_object (fp.get_currentaxes ());
 		if (ax && ax.isa ("axes")) 
 		  {
 		    axes::properties& ap = 
 		      dynamic_cast<axes::properties&> (ax.get_properties ());
 		    pixel2pos (px0, py0, x0, y0);
 		    pixel2pos (Fl::event_x (), Fl::event_y (), x1, y1);
-		    Matrix pp (1,2,0);
+		    Matrix xl (1,2,0);
+		    Matrix yl (1,2,0);
 		    if (x0 < x1)
 		      {
-			pp(0) = x0;
-			pp(1) = x1;
+			xl(0) = x0;
+			xl(1) = x1;
 		      }
 		    else
 		      {
-			pp(0) = x1;
-			pp(1) = x0;
+			xl(0) = x1;
+			xl(1) = x0;
 		      }
-		    ap.set_xlim (pp);
 		    if (y0 < y1)
 		      {
-			pp(0) = y0;
-			pp(1) = y1;
+			yl(0) = y0;
+			yl(1) = y1;
 		      }
 		    else
 		      {
-			pp(0) = y1;
-			pp(1) = y0;
+			yl(0) = y1;
+			yl(1) = y0;
 		      }
-		    ap.set_ylim (pp);
+		    ap.zoom (xl, yl);
 		    mark_modified ();
 		  }
 	      }
 	    // one click -- select axes
 	    else if ( Fl::event_clicks () == 0)
 	      {
 		std::cout << "ca="<< h0.value ()<<"\n";
 		if (h0.ok ())
 		  fp.set_currentaxes (h0.value());
 		return 1;
 	      }
 	  }
 	else if (Fl::event_button () == 3)
 	  {
-	    axis_auto ();
+	    graphics_object ax = 
+	      gh_manager::get_object (fp.get_currentaxes ());
+	    if (ax && ax.isa ("axes")) 
+	      {
+		axes::properties& ap = 
+		  dynamic_cast<axes::properties&> (ax.get_properties ());
+		ap.unzoom ();
+		mark_modified ();
+	      }
 	  }
 	break;
       }
 
     return retval;
   }
 
 };
